### ISSUE TITLE
Pre-Re instances cooldown check

### ISSUE THREAD

**Issue by Ridley8819** [HerculesWS/Hercules#1582]
Ok so I had a longer discussion with @iRagno regarding instances, and the way we do check for CD in athena, how its done on iRO, and how its done on kRO. So we thought to have a new battleflag in features.conf to offer the following behaviour for instances.

--> athena behaviour = check CD on party leader, allow leader switch (bug?) 
--> iro behaviour = CD quest got applied to every party member, but it doesn't get checked until the instance is completed/destroyed. This allows party member who disconnect, for whatever reason, to rejoin.
--> kro behaviour (will not be implemented immediately since we're missing dialogues for it) = instant apply CD on all party member, no CD applied

This would fix the leader switch bug for all pre re instances, as well as allowing disconnecting party member to reenter in time.

When talking with @iRagno, he was excited to take this task.

**Comment by Mysteries** [comment]
I agree. I think since we're all use to it... we should add it as a setting in the features.conf =/

**Comment by iRagno** [comment]
About new behaviour, that is, disable re-entering the instance when player comes out from instance map, I think we can use actual messages for when cool down has been seted, until having official dialogues that requires ingame test.

In that way, we can bring an option asap according to the needs.

**Comment by AnisotropicDefixation** [comment]
Some of the new instances instead add a quest entry that allows you to reenter freely for the first 5 or so minutes.

**Comment by Asheraf** [comment]
he is talking about `Pre-Re instances`

**Comment by iRagno** [comment]
Added pull request with fix to entrance exploit in Endless Tower, Nydhogur Nest and Orcs Memorial instance dungeons.

Sealed Shrine has disabled reentering instance once was created and needs more investigation if that should be default behaviour. I assume it could be fixed with this, but messages are kind of weird, needs to confirm behaviour with officials.
